src/logic/calculate.js

```diff
@@ -0,0 +1,119 @@
+import operate from './operator';
+
+function isNumber(item) {
+  return !!item.match(/[0-9]+/);
+}
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
```

Comment:
-  [ ] Please write tests that cover all the functionalities & scenarios of the operate.js and calculate.js modules.